dbt/adapters/athena/impl.py

```diff
@@ -494,3 +523,31 @@ def persist_docs_to_glue(
                     col_obj["Comment"] = clean_sql_comment(col_comment)
 
         glue_client.update_table(DatabaseName=relation.schema, TableInput=updated_table)
+
+    @available
+    def list_schemas(self, database: str) -> List[str]:
+        conn = self.connections.get_thread_connection()
+        client = conn.handle
+
+        with boto3_client_lock:
+            glue_client = client.session.client("glue", region_name=client.region_name, config=get_boto3_config())
+
+        paginator = glue_client.get_paginator("get_databases")
+        result = []
+        for page in paginator.paginate():
+            for db in page["DatabaseList"]:
+                result.append(db["Name"])
+        return result
+
+    @available
+    def get_columns_in_relation(self, relation: AthenaRelation) -> List[Column]:
```

Comment:
Maybe deleting tables via Glue API too?

```python
    @available
    def delete_table(self, relation: AthenaRelation):
        conn = self.connections.get_thread_connection()
        client = conn.handle
        with boto3_client_lock:
            glue_client = client.session.client("glue", region_name=client.region_name)

        try:
            glue_client.delete_table(DatabaseName=relation.schema, Name=relation.identifier)
        except ClientError as e:
            if e.response['Error']['Code'] == 'EntityNotFoundException':
                logger.debug("Table '{}' does not exists - Ignoring", relation)
            else:
                raise
```


Then change `athena__drop_relation` to `{%- do adapter.delete_table(relation) -%}`

Reply:
We cannot drop iceberg tables via glue api, as they are managed in athena

Reply:
Also, my recommendation is to cover only information schema replacement in this PR, and consider usage of glue apis for others ddl and dml operations

Reply:
I was about to create a PR for the `glue_client.delete_table` change, but looks like @Jrmyy, you got first 🥇 

Actually I would welcome this change for Iceberg tables since according to AWS Support they do suggest to drop the iceberg table using the procedure delete from s3 + delete from glue, since unfortunately the `DROP TABLE` statement has known bugs for buckets containing a large amount of files, which they meant can't be resolved soon since it involves an architecture change.

Reply:
I will follow the recommendation of @nicor88 and keep the PR for metadata only.
But let's figure out in an other PR what we want to do for delete.

Reply:
@aartaria good point for iceberg, where is it mentioned about the issues with drop stetements for iceberg? I might have missed it.